dashpay/src/main/kotlin/org/dashj/platform/contracts/wallet/TxMetadataItem.kt

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2022-present, Dash Core Group
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package org.dashj.platform.contracts.wallet
+
+import com.google.common.primitives.Ints
+import org.dashj.platform.dpp.toHex
+
+/**
+ * Transaction metadata item
+ *
+ * @property txId The transaction id (hash) in big endian format
+ * @property memo
+ * @property exchangeRate
+ * @property currencyCode
+ * @property taxCategory
+ * @property service
+ * @constructor Create empty Tx metadata item
+ */
+
+class TxMetadataItem(
+    val txId: ByteArray,
+    val memo: String? = null,
+    val exchangeRate: Double? = null,
+    val currencyCode: String? = null,
+    val taxCategory: String? = null,
+    val service: String? = null,
+    val version: Int = 0
+) {
+    val data = hashMapOf<String, Any?>()
+
+    constructor(rawObject: Map<String, Any?>) :
+    this(
+        rawObject["txId"] as ByteArray,
+        rawObject["memo"] as? String,
+        rawObject["exchangeRate"] as? Double,
+        rawObject["currencyCode"] as? String,
+        rawObject["taxCategory"] as? String,
+        rawObject["service"] as? String,
+        rawObject["version"] as Int
+    ) {
+        data.putAll(rawObject)
+    }
+
+    fun toObject(): Map<String, Any?> {
+        val map = hashMapOf<String, Any?>(
+            "txId" to txId,
+            "version" to version
+        )
+
+        memo?.let {
+            map["memo"] = it
+        }
+
+        exchangeRate?.let {
+            map["exchangeRate"] = it
+        }
+
+        currencyCode?.let {
+            map["currencyCode"] = it
+        }
+
+        taxCategory?.let {
+            map["taxCategory"] = it
+        }
+
+        service?.let {
+            map["service"] = it
+        }
+
+        return map
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) {
+            return true
+        } else if (other is TxMetadataItem) {
+            return txId.contentEquals(other.txId) &&
+                version == other.version
+            memo == other.memo &&
+                exchangeRate == other.exchangeRate &&
+                currencyCode == other.currencyCode &&
+                taxCategory == other.taxCategory &&
+                service == other.service
+        }
+        return false
```

Comment:
This class looks like it could be a `data class`. With that change, the equality check, hash code and `toString()` function can be generated automatically by Kotlin.

Reply:
This is a good point.  I can make this change in the next Story for Tx Metadata